### PR TITLE
feat(connector): add Xiaomi social login connector

### PR DESCRIPTION
**Pull request by u0x01**
## Summary

Added Xiaomi social connector to enable users to sign in to Logto using their Xiaomi accounts. Key features include:

- Implemented Xiaomi OAuth 2.0 authorization flow
- Support fetching user profile information (avatar, nickname, etc.)
- Configurable authorization scope and skip xiaomi auth confirmation page option
- Comprehensive unit test coverage

## Testing

- Added complete unit tests covering:
  - Get authorization URI
  - Exchange access token
  - Fetch user info
  - Error handling scenarios
- Manual testing completed in local environment to verify the full authorization flow

## Checklist

- [x] `.changeset` 
- [x] unit tests
- [x] integration tests - N/A for this connector
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/connectors/connector-xiaomi/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Xiaomi connector
+
+Logto 官方小米社交登录连接器。
```
Comment (charIeszhao) [inline packages/connectors/connector-xiaomi/README.md:3]: Please translate your README into English.

File: packages/connectors/connector-xiaomi/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+| 12005 | 使用健康心电图服务 | 健康 |
+| 16000 | 获取小米卡包卡券 | app/get_pass |
+| 20000 | 启用小爱智能语音服务 | 小爱 |
+| 40000 | 启用云端 AI 服务 | 内部使用 |
+
+可以通过空格分隔配置多个权限范围，例如: `1 3 6000`。
+
+## 测试小米连接器
+
+配置完成。别忘了在[登录体验](https://docs.logto.io/zh-cn/docs/recipes/configure-connectors/social-connector#%E5%90%AF%E7%94%A8%E7%A4%BE%E4%BA%A4%E7%99%BB%E5%BD%95)中启用该连接器。
```
Comment (charIeszhao) [inline packages/connectors/connector-xiaomi/README.md:57]: It seems the link you are using here is out-of-date.

Is this the same one you are refering to? https://docs.logto.io/zh-CN/connectors/social-connectors#enable-social-sign-in

**Review by charIeszhao — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

File: packages/connectors/connector-xiaomi/src/index.test.ts [hunk 3]
```diff
@@ -78,11 +84,62 @@ describe('getUserInfo', () => {
...
-  it('throws SocialAccessTokenInvalid error if remote response code is 401', async () => {
-    nock(userInfoEndpoint).get('').query(true).reply(401);
+  it('throws SocialAccessTokenInvalid error if remote response code is 403', async () => {
+    // Mock the token endpoint to return a valid token first
+    nock(accessTokenEndpoint)
+      .post('')
+      .reply(
+        200,
+        `&&&START&&&${JSON.stringify({
+          access_token: 'invalid_token',
```
Comment (charIeszhao) [inline packages/connectors/connector-xiaomi/src/index.test.ts:94]: Should be `valid_token` maybe?

Comment (charIeszhao) [reply packages/connectors/connector-xiaomi/src/index.test.ts:94]: Why do we need to return a valid token first?

**Pull request merged**